From 6855847dea21094066a37ddb2d765665059c0b64 Mon Sep 17 00:00:00 2001
From: Sidhanta Sahu <quic_sidhanta@quicinc.com>
Date: Mon, 22 Jul 2024 19:55:45 -0700
Subject: [PATCH] wifi: ath12k: Add null check to prevent crash in
 ath12k_dp_tx_put_bank_profile

Currently, ath12k_dp_free is called during recovery path which frees up dp->bank_profiles.


Direct access to dp->bank_profiles in ath12k_dp_tx_put_bank_profile now includes
a null check. This mitigates a potential use-after-free scenario during
recovery when dp->bank_profiles might be freed, and a userspace crash or mac
remove interface operation could call ath12k_dp_tx_put_bank_profile.

During recovery dp->bank_profiles are freed.

Patch-Dependency: 707-b-ath12k-add-support-for-MLO-SSR-Mode0.patch 
Patch-Work: None

Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp.c b/drivers/net/wireless/ath/ath12k/dp.c
index 346e9d4..58f987c 100644
--- a/drivers/net/wireless/ath/ath12k/dp.c
+++ b/drivers/net/wireless/ath/ath12k/dp.c
@@ -741,6 +741,9 @@ inc_ref_and_return:
 
 void ath12k_dp_tx_put_bank_profile(struct ath12k_dp *dp, u8 bank_id)
 {
+	if (!dp->bank_profiles)
+		return;
+
 	spin_lock_bh(&dp->tx_bank_lock);
 	dp->bank_profiles[bank_id].num_users--;
 	spin_unlock_bh(&dp->tx_bank_lock);
-- 
2.17.1

